src/nvim/ex_getln.c

```diff
@@ -4561,8 +4575,9 @@ expand_cmdline (
     return EXPAND_NOTHING;
   }
 
-  /* add star to file name, or convert to regexp if not exp. files. */
-  xp->xp_pattern_len = (int)(str + col - xp->xp_pattern);
+  // add star to file name, or convert to regexp if not exp. files.
+  assert(str + col - xp->xp_pattern >= 0);
+  xp->xp_pattern_len = (size_t)(str + col - xp->xp_pattern);
```

Comment:
PVS v1028

Reply:
mixture of ints and pointers, will rewrite for clarity.